tracer/test/Datadog.Trace.Tests/Propagators/MultiSpanContextPropagatorTests.cs

```diff
@@ -798,6 +799,29 @@ public void TraceContextPrecedence_ConsistentBehaviour_WithDifferentTraceIds(boo
                        });
         }
 
+        [Theory]
+        [InlineData("dd=s:2;p:000000003ade68b1,foo=1", "1", "987654321", 987654321, null)]
+        [InlineData("dd=s:2;p:000000000000000a,foo=1", "2", "10", 10, null)]
+        [InlineData("dd=s:2;p:000000000000000a,foo=1", "1", "10", 987654321, "000000000000000a")]
+        [InlineData("dd=s:2,foo=1", "1", "10", 987654321, "000000000000000a")]
+        [InlineData("dd=s:2;p:8fffffffffffffff,foo=1", "1", "10", 987654321, "8fffffffffffffff")]
+        public void Datadog_W3C_TraceState_ParentExtracted(string tracestate, string traceId, string parentId, ulong expectedSpanId, string expectedParentTag)
```

Comment:
To make sure this is thoroughly unit-tested, can you also run these tests in the following ways:

1. If we switch all the Datadog headers to their B3Multi form and we switch the propagator configuration from Datadog to B3Multi, what happens? I expect both the SpanId and LastParentId to remain the same
2. If we configure the propagators to be Datadog then B3Multi and we update the headers to use those two formats, what happens? In this case, we should _only_ be using the Datadog information because our logic should only update the SpanId and LastParentId for the `tracecontext` propagator

I recognize that this testing is more than what our W3C Phase 3 system-tests do today, but I anticipate that these may be issues in the current implementation so I'd like to get ahead of the issues now so we don't ship some bad behaviors.

Reply:
Did some changes based on the configuration below: https://github.com/DataDog/dd-trace-dotnet/pull/5721/commits/aaac0d0d30130b6723f8528bf209055b09569792